### PR TITLE
Add scope filter to stats window

### PR DESCRIPTION
**Pull request by TimonPost**
### Checklist

* [x] I have read the [Contributor Guide](../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

Adds a scope filter, similar to flamegraph filter, to the stats panel. 

- Moves the `Filter` component out of the flamegraph file.
- Implements the `Filter` component for the stats file to filter scopes.

### Demo
image

### CODE REVIEW

File: puffin_egui/src/flamegraph.rs [hunk 1]
```diff
@@ -159,7 +125,7 @@ pub struct Options {
     pub flamegraph_threads: IndexMap<String, ThreadVisualizationSettings>,
 
     #[cfg_attr(feature = "serde", serde(skip))]
-    filter: Filter,
+    pub filter: Filter,
```
Comment (NiklasNummelin) [inline puffin_egui/src/flamegraph.rs:128]: `pub (crate)`?

File: puffin_egui/src/stats.rs [hunk 2]
```diff
@@ -1,6 +1,16 @@
 use puffin::*;
 
-pub fn ui(ui: &mut egui::Ui, frames: &[std::sync::Arc<UnpackedFrameData>]) {
+use crate::filter::Filter;
+
+#[derive(Clone, Debug, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(default))]
+pub struct Options {
```
Comment (NiklasNummelin) [inline puffin_egui/src/stats.rs:8]: Why allow to serialize and deserialize this if the only field in it is skipped?

**Comment by TimonPost** [comment]
I am not sure how serialization works. The current options seem to implement serde but they dont seem to be saved anywere.

File: puffin_egui/src/lib.rs [hunk 3]
```diff
@@ -324,7 +326,8 @@ impl Default for View {
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 #[cfg_attr(feature = "serde", serde(default))]
 pub struct ProfilerUi {
-    pub options: flamegraph::Options,
+    pub flamegraph_options: flamegraph::Options,
```
[the remark this answers is not in the document]
Comment (TimonPost) [reply puffin_egui/src/lib.rs:329]: not sure if its serialized 🤔

[the remark this answers is not in the document]
Comment (NiklasNummelin) [reply puffin_egui/src/lib.rs:329]: It's up to the user of `ProfilerUI` :)

**Pull request merged**